py/legacyanalysis/sersic-agn.py

```diff
@@ -0,0 +1,269 @@
+        tr = Tractor(tims, [devsrc])
+        tr.freezeParam('images')
+        tr.optimize_loop()
+        print('Fit DeV source:', devsrc)
+        devmods = list(tr.getModelImages())
+        showboth(tims, devmods, wcs, bands);
+        s = flux_string(devsrc.brightness)
+        plt.suptitle('DeV model: ' + s + '\ndchisq 0.')
+        plt.savefig('src%02i-dev.png' % isrc)
+        devchi = 2. * tr.getLogLikelihood()
+        
+        dasrc = DevAgnGalaxy(devsrc.pos.copy(), devsrc.brightness.copy(), devsrc.shape.copy(), NanoMaggies(**dict([(b,1.) for b in bands])))
+        #dasrc = DevAgnGalaxy(RaDecPos(ra,dec), NanoMaggies(**dict([(b,10.) for b in bands])), EllipseESoft(0., 0., 0.), NanoMaggies(**dict([(b,1.) for b in bands])))
+        tr = Tractor(tims, [dasrc])
+        tr.freezeParam('images')
+        tr.optimize_loop()
+        print('Fit Dev+PSF source:', dasrc)
+        damods = list(tr.getModelImages())
+        showboth(tims, damods, wcs, bands)
+        s1 = flux_string(dasrc.brightnessDev)
+        s2 = flux_string(dasrc.brightnessPsf)
+        pcts = [100. * dasrc.brightnessPsf.getFlux(b) / dasrc.brightnessDev.getFlux(b) for b in bands]
+        s3 = ', '.join(['%.2f' % p for p in pcts])
+        dachi = 2. * tr.getLogLikelihood()
+        plt.suptitle('DeV + Point Source model: DeV %s, PSF %s' % (s1, s2) + ' (%s %%)' % s3 +
+                     '\ndchisq %.1f' % (dachi - devchi))
+        plt.savefig('src%02i-devcore.png' % isrc)
+    
+        #sersrc = SersicGalaxy(RaDecPos(ra, dec), NanoMaggies(**dict([(b,10.) for b in bands])), EllipseESoft(0.5, 0., 0.), SersicIndex(4.0))
+        sersrc = SersicGalaxy(devsrc.pos.copy(), devsrc.brightness.copy(), devsrc.shape.copy(), SersicIndex(4.0))
+        tr = Tractor(tims, [sersrc])
+        tr.freezeParam('images')
+        r = tr.optimize_loop()
+        print('Opt:', r)
+        print('Fit Ser source:', sersrc)
+        if sersrc.sersicindex.getValue() >= 6.0:
+            sersrc.freezeParam('sersicindex')
+            r = tr.optimize_loop()
+            print('Re-fit Ser source:', sersrc)
+        sermods = list(tr.getModelImages())
+        showboth(tims, sermods, wcs, bands)
+        s = flux_string(sersrc.brightness)
+        serchi = 2. * tr.getLogLikelihood()
+        plt.suptitle('Sersic model: %s, index %.2f' % (s, sersrc.sersicindex.getValue()) +
+                    '\ndchisq %.1f' % (serchi - devchi))
+        plt.savefig('src%02i-ser.png' % isrc)
+    
+        #sasrc = SersicAgnGalaxy(RaDecPos(ra, dec), NanoMaggies(**dict([(b,10.) for b in bands])), EllipseESoft(0.5, 0., 0.), SersicIndex(4.0), NanoMaggies(**dict([(b,1.) for b in bands])))
+
+        si = sersrc.sersicindex.getValue()
+        if si > 6.0:
+            si = 4.0
+        si = SersicIndex(si)
+        
+        sasrc = SersicAgnGalaxy(sersrc.pos.copy(), sersrc.brightness.copy(), sersrc.shape.copy(), si, NanoMaggies(**dict([(b,1.) for b in bands])))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['NanoMaggies' may be undefined, or defined from star imports: legacypipe.survey, tractor, tractor.devagn, tractor.galaxy](https://app.codacy.com/app/dstndstn/legacypipe/pullRequest?prid=4730106)